### PR TITLE
Fix change of user with the same role.

### PR DESCRIPTION
**Pull request by pvalsecc**
When doing a logout and login with a different user within the same
group, the "Logged in as" text on the top right corner was still showing
the previous username.

That was because the edit.js, view.js, ... pages were having role= added
as parameter to avoid the cache, not the username. So the page was not
reloaded by the browser and the username sitting in those files was
remaining at the old value.

The fix is to add username= instead of role= to the URLs.

### CODE REVIEW

File: c2cgeoportal/lib/__init__.py [hunk 1]
```diff
@@ -152,8 +152,11 @@ def __call__(self, request, elements, kw):
             from c2cgeoportal.lib.cacheversion import get_cache_version
             query["cache_version"] = get_cache_version()
 
-        if self.role and request.user and request.user.role:
-            query["role"] = request.user.role.id
+        if self.role and request.user:
```
Comment (sbrunner) [inline c2cgeoportal/lib/__init__.py]: I think that the `self.role` can also be removed...

Comment (pvalsecc) [inline c2cgeoportal/lib/__init__.py]: I don't think we should remove it. In some cases (one, actually), we don't want to invalidate the cache if the user changes identity. This one:
```python
    config.add_route(
        "layers_enumerate_attribute_values",
        "/layers/{layer_name}/values/{field_name}",
        request_method="GET",
        pregenerator=C2CPregenerator(),
    )
```

**Comment by sbrunner** [comment]
Thanks, merge :-)

File: c2cgeoportal/lib/__init__.py (hunk 1 above)
Comment (sbrunner) [inline c2cgeoportal/lib/__init__.py]: Effectively, thanks :-)

**Pull request merged**